bftengine/tests/messages/PrePrepareMsg_test.cpp

```diff
@@ -58,6 +103,53 @@ class PrePrepareMsgTestFixture : public ::testing::Test {
   std::unique_ptr<SigManager> sigManager;
 };
 
+TEST_F(PrePrepareMsgTestFixture, finalize_and_validate) {
+  ReplicasInfo replicaInfo(createReplicaConfig(), false, false);
+
+  ReplicaId senderId = 1u;
+  ViewNum viewNum = 2u;
+  SeqNum seqNum = 3u;
+  CommitPath commitPath = CommitPath::OPTIMISTIC_FAST;
+  const char rawSpanContext[] = {"span_\0context"};
+  const std::string spanContext{rawSpanContext, sizeof(rawSpanContext)};
+  std::vector<std::shared_ptr<ClientRequestMsg>> crmv;
+  create_random_client_requests(crmv, 50u);
+  size_t req_size = 0;
+  for (const auto& crm : crmv) {
+    req_size += crm->size();
+  }
+
+  std::cout << "Total size : " << req_size << std::endl;
+
+  PrePrepareMsg msg(senderId, viewNum, seqNum, commitPath, concordUtils::SpanContext{spanContext}, req_size);
+
+  EXPECT_EQ(msg.viewNumber(), viewNum);
+  EXPECT_EQ(msg.seqNumber(), seqNum);
+  EXPECT_EQ(msg.firstPath(), commitPath);
+  EXPECT_EQ(msg.isNull(), false);
+  EXPECT_EQ(msg.numberOfRequests(), 0u);
+
+  std::vector<std::string> dv;
+  for (const auto& crm : crmv) {
+    std::cout << "message size : " << crm->size() << std::endl;
+    msg.addRequest(crm->body(), crm->size());
+    Digest d;
+    DigestUtil::compute(crm->body(), crm->size(), (char*)&d, sizeof(Digest));
+    dv.push_back({d.content(), sizeof(Digest)});
+  }
+  EXPECT_NO_THROW(msg.finishAddingRequests());  // create the digest
+  EXPECT_EQ(msg.numberOfRequests(), 50u);
+
+  std::string dod;
+  for (const auto& s : dv) {
+    dod.append(s);
+  }
+  Digest d;
+  DigestUtil::compute(dod.c_str(), dod.size(), (char*)&d, sizeof(Digest));
+  EXPECT_EQ(d == msg.digestOfRequests(), true);
```

Comment:
This test should be `EXPECT_EQ(d, msg.digestOfRequests())`. Otherwise, you will just get an error of "false == true" rather than a precise printing of what went wrong.